lib/charms/mongodb/v1/mongodb_tls.py

```diff
@@ -365,3 +450,27 @@ def _get_subject_name(self) -> str:
             return self.charm.get_config_server_name() or self.charm.app.name
 
         return self.charm.app.name
+
+    def is_set_waiting_for_cert_to_update(
+        self,
+        internal=False,
+    ) -> bool:
+        """Returns True we are waiting for a cert to update."""
+        if internal:
+            json.loads(self.charm.app_peer_data.get(WAIT_CERT_UPDATE, "false"))
+        else:
+            json.loads(self.charm.unit_peer_data.get(WAIT_CERT_UPDATE, "false"))
+
+    def set_waiting_for_cert_to_update(
+        self,
+        waiting: bool,
+        internal: bool,
+    ):
+        """Sets a boolean indicator, for whether or not we are waiting for a cert to update."""
+        if internal and not self.charm.unit.is_leader():
+            return
+
+        if internal:
+            self.charm.app_peer_data[WAIT_CERT_UPDATE] = json.dumps(waiting)
+        else:
+            self.charm.unit_peer_data[WAIT_CERT_UPDATE] = json.dumps(waiting)
```

Comment:
```suggestion
        if internal and self.charm.unit.is_leader():
            self.charm.app_peer_data[WAIT_CERT_UPDATE] = json.dumps(waiting)
        elif not internal:
            self.charm.unit_peer_data[WAIT_CERT_UPDATE] = json.dumps(waiting)
```

Reply:
maybe its that its 17h00 and my coffee is wearing off, but I find this suggestion a little hard to read (even though it is correct)

I am going to close it for now. If you come back to this and feel strongly let me know